src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/FilePathNormalizer.cs

```diff
@@ -19,9 +21,13 @@ internal static class FilePathNormalizer
 
     private class FilePathNormalizingComparer : IEqualityComparer<string>
     {
+        private Func<char, char> _charConverter = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+            ? c => c
+            : char.ToLowerInvariant;
+
         public bool Equals(string? x, string? y) => FilePathNormalizer.FilePathsEquivalent(x, y);
 
-        public int GetHashCode([DisallowNull] string obj) => FilePathNormalizer.GetHashCode(obj);
+        public int GetHashCode([DisallowNull] string obj) => FilePathNormalizer.GetHashCode(obj, _charConverter);
```

Comment:
I don't think `[DisallowNull]` is needed here.

Reply:
Its on `IEqualityComparer` so Roslyn put it there when implementing the interface 🤷‍♂️